lib/ret_web/plugs/portal_header_authorization.ex

```diff
@@ -0,0 +1,16 @@
+defmodule RetWeb.Plugs.PortalHeaderAuthorization do
+  import Plug.Conn
+  @header_name "x-ret-portal-access-key"
+
+  def init(default), do: default
+
+  def call(conn, _default) do
+    expected_value = Application.get_env(:ret, RetWeb.Plugs.PortalHeaderAuthorization)[:portal_access_key]
+
+    case conn |> get_req_header(@header_name) do
+      [""] -> conn |> send_resp(401, "") |> halt()
```

Comment:
Is this to catch the case where portal_access_key is miss-configured and left blank?

Reply:
Yes, we didn't want the header with an empty string value to pass.